File: VERSION1.0/ClientEngine/System/src/JavaMethod.cpp
//
//  JavaMethod.m
//  cocotest
//
//  Created by xiezhenghai on 10-11-24.
//  Copyright 2010 (ÍøÁú)DeNA. All rights reserved.
//

#include "JavaMethod.h"
#include "Utility.h"

int FileOp::readByte(FILE* f)
{
	if (!f)
	{
		NDAsssert(0);
		return 0;
	}

	size_t size = sizeof(unsigned char);
	unsigned char ucData = 0;
	size_t read = fread(&ucData, 1, size, f);
	if (size != read)
	{
		NDAsssert(0);
		return 0;
	}
	return ucData;
}

int FileOp::readShort(FILE* f)
{
	if (!f)
	{
		NDAsssert(0);
		return 0;
	}

	size_t size = sizeof(unsigned short);
	unsigned short usData = 0;
	unsigned char shortBuf[2] = {0x00};
	size_t read = fread(&shortBuf, 1, size, f);
	if (size != read)
	{
		NDAsssert(0);
		return 0;
	}
	usData = ((unsigned short)shortBuf[0] << 8) + shortBuf[1];
	return usData;
}

int FileOp::readInt(FILE* f)
{
	if (!f)
	{
		NDAsssert(0);
		return 0;
	}

	size_t size = sizeof(unsigned int);
	unsigned int unData = 0;
	unsigned char intBuf[4] = {0x00};
	size_t read = fread(&intBuf, 1, size, f);
	if (size != read)
	{
		NDAsssert(0);
		return 0;
	}
	unData = ((unsigned int)intBuf[0] << 24) +
			((unsigned int)intBuf[1] << 16) +
			((unsigned int)intBuf[2] << 8) +
			((unsigned int)intBuf[3]);
	return unData;
}

std::string FileOp::readUTF8String(FILE* f)
{
	if (!f)
	{
		NDAsssert(0);
		return "";
	}

	unsigned char ucHight	= this->readByte(f);
	unsigned char ucLow		= this->readByte(f);
	
	size_t strlen = (unsigned int)(ucHight << 8) + ucLow;
	if (strlen == 0)
	{
		return "";
	}

	unsigned char pszTemp[4096] = {0};
	size_t read = fread(&pszTemp, 1, strlen, f);
	if (read != strlen)
	{
		NDAsssert(0);
		return "";
	}

	return (char*)&pszTemp;
}

std::string FileOp::readUTF8StringNoExcept(FILE* f)
{
	if (!f)
	{
		return "";
	}

	unsigned char ucHight	= this->readByte(f);
	unsigned char ucLow		= this->readByte(f);

	size_t strlen = (unsigned int)(ucHight << 8) + ucLow;
	if (strlen == 0)
	{
		return "";
	}

	unsigned char tmp[4096] = {0};
	size_t read = fread(&tmp, 1, strlen, f);
	if (read != strlen)
	{
		return "";
	}

	return (char*)&tmp;
}

File: old_bak/SMYS_CLIENT_IPHONE/Classes/Engine/System/src/JavaMethod.m
//
//  JavaMethod.m
//  cocotest
//
//  Created by xiezhenghai on 10-11-24.
//  Copyright 2010 __MyCompanyName__. All rights reserved.
//

#import "JavaMethod.h"


@implementation NSInputStream(ND)

- (int)readByte
{
	static unsigned char byteBuf[1] = {0x00};
	int readLen = [self read:byteBuf maxLength:1];
	if (readLen) 
	{
		return byteBuf[0];
	}
	else 
	{
		NDAsssert(0);
		return 0;
		//@throw [NSException exceptionWithName:@"NSInputStream" reason:@"read from stream error" userInfo:nil];
	}
}

- (int)readShort
{
	static unsigned char shortBuf[2] = {0x00};
	int readLen = [self read:shortBuf maxLength:2];
	if (readLen) 
	{
		return (short)((shortBuf[0] << 8) + shortBuf[1]);
	}
	else 
	{
		NDAsssert(0);
		return 0;
		//@throw [NSException exceptionWithName:@"NSInputStream" reason:@"read from stream error" userInfo:nil];
	}
}

- (int)readInt
{
	static unsigned char intBuf[4] = {0x00};
	int readLen = [self read:intBuf maxLength:4];
	if (readLen) 
	{
		return ((intBuf[0] << 24) + (intBuf[1] << 16) + (intBuf[2] << 8) + intBuf[3]);
	}
	else 
	{
		NDAsssert(0);
		return 0;
		//@throw [NSException exceptionWithName:@"NSInputStream" reason:@"read from stream error" userInfo:nil];
	}
}

- (NSString *)readUTF8String
{
	NSString* str = @"";
	int strLen = ([self readByte] << 8) + [self readByte];	
	if (strLen) 
	{
		unsigned char *buffer = (unsigned char *)malloc(sizeof(unsigned char) * strLen + 1);
		memset(buffer, 0x00, strLen + 1);
		
		[self read:buffer maxLength:strLen];
		
		str = [NSString stringWithUTF8String:(const char*)buffer];
		
		free(buffer);
	}
	
	return str;
}

- (NSString *)readUTF8StringNoExcept
{
	unsigned char byteBufHight, byteBufLow;
	int readLen = [self read:&byteBufHight maxLength:1];
	if (!readLen) 
	{
		return @"";
	}
	
	readLen = [self read:&byteBufLow maxLength:1];
	if (!readLen) 
	{
		return @"";
	}
	
	int strLen = ((int)byteBufHight << 8) + byteBufLow;
	
	unsigned char *buffer = (unsigned char *)malloc(sizeof(unsigned char) * strLen + 1);
	memset(buffer, 0x00, strLen + 1);
	
	readLen = [self read:buffer maxLength:strLen];
	NSString* str = @"";
	if (readLen) 
	{
		str = [NSString stringWithUTF8String:(const char*)buffer];
	}
	
	free(buffer);
	return str;
}

@end
